source/upgrade.txt

```diff
@@ -0,0 +1,156 @@
+.. _java-upgrade-driver:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+Overview
+--------
+
+In this section, you can identify the changes you need to make to your
+application to upgrade your driver to a new version or to transition from
+the legacy API to the current API.
+
+When you upgrade your driver version or use multiple versions of the
+driver, if there are any breaking changes between them that you have not
+addressed in the :ref:`Breaking Changes <java-breaking-changes>` section,
+you may encounter one or more of the following issues:
+
+- Your application may not compile.
+- Your application may not be compatible with data that originated from an
```

Comment:
This is scary.  What change is this referring to?

Reply:
This is related to [JAVA-3854](https://jira.mongodb.org/browse/JAVA-3854). The issue report mentions incompatible serialization ids when attempting to use two different driver versions to serialize/deserialize an object that contains an ObjectId.
It's also what I was attempting to communicate (here)[https://github.com/mongodb/docs-java/pull/212#discussion_r824237100].

Reply:
It just makes it sound a lot scarier than it actually is.  The serialization incompatibility in JAVA-3854 is not about serializing ObjectId instances to and from BSON, which would be absolutely horrible.  It's about serializing to and from Java's Object Serialization format, which is a much, much smaller group of people.  So I think including it here as a general statement will do more harm than good.

Reply:
I know this has to do with Java's object serialization and agree it has to do with a smaller audience, hence my comment here: https://github.com/mongodb/docs-java/pull/212#discussion_r824845764

Reply:
OK.  I still think it shouldn't be mentioned here, as it could easily be minconstrued.

Reply:
I agree and removed it in this commit:
https://github.com/mongodb/docs-java/pull/212/commits/05fea111cd793956a1a46f862b9069aa8359dfde